nimare/workflows/macm.py

```diff
@@ -1,12 +1,157 @@
+               short_help='Run a meta-analytic coactivation modeling (MACM) '
+                          'analysis using activation likelihood estimation '
+                          '(ALE) on a NiMARE dataset file and a target mask.')
+@click.argument('dataset_file', type=click.Path(exists=True))
+@click.option('--mask', '--mask_file', type=click.Path(exists=True))
+@click.option('--output_dir', help="Where to put the output maps.")
+@click.option('--prefix', help="Common prefix for output maps.")
+@click.option('--n_iters', default=N_ITERS_DEFAULT, show_default=True,
+              help="Number of iterations for permutation testing.")
+@click.option('--v_thr', default=CLUSTER_FORMING_THRESHOLD_P_DEFAULT,
+              show_default=True,
+              help="Voxel p-value threshold used to create clusters.")
+@click.option('--c_thr', default=CLUSTER_SIZE_THRESHOLD_Q_DEFAULT,
+              show_default=True,
+              help="Cluster size corrected p-value threshold.")
+@click.option('--n_cores', default=-1,
+              show_default=True,
+              help="Number of processes to use for meta-analysis. If -1, use "
+                   "all available cores.")
+def macm_workflow(dataset_file, mask_file, output_dir=None,
+                  prefix=None,
+                  n_iters=N_ITERS_DEFAULT,
+                  v_thr=CLUSTER_FORMING_THRESHOLD_P_DEFAULT,
+                  c_thr=CLUSTER_SIZE_THRESHOLD_Q_DEFAULT,
+                  n_cores=-1):
+    """
+    Perform MACM with ALE algorithm.
+    """
+    click.echo("Loading coordinates...")
+    dset = Dataset(dataset_file)
+    sel_ids = dset.get_studies_by_mask(mask_file)
+
+    # override sample size
+    n_subs_db = dset.coordinates.drop_duplicates('id')['n'].astype(float).astype(int).sum()
+    sel_coords = dset.coordinates.loc[dset.coordinates['id'].isin(sel_ids)]
+    n_subs_sel = sel_coords.drop_duplicates('id')['n'].astype(float).astype(int).sum()
+    click.echo("{0} studies selected out of {1}.".format(len(sel_ids),
+                                                         len(dset.ids)))
+
+    boilerplate = """
+Meta-analytic connectivity modeling (MACM; Laird et al., 2009; Robinson et al.,
+2009; Eickhoff et al., 2010) analysis was performed with the activation
+likelihood estimation (ALE; Turkeltaub, Eden, Jones, & Zeffiro, 2002; Eickhoff,
+Bzdok, Laird, Kurth, & Fox, 2012; Turkeltaub et al., 2012) meta-analysis
+algorithm using NiMARE. The input database included {n_foci_db}
+foci from {n_subs_db} participants across {n_exps_db} studies/experiments, from
+which studies/experiments were selected for analysis if they had at least one
+focus inside the target mask. The resulting sample included {n_foci_sel}
+foci from {n_subs_sel} participants across {n_exps_sel} studies/experiments.
+
+Foci were convolved with Gaussian kernels determined by sample size,
+implemented on the MNI 152 template (Fonov et al., 2009; Fonov et al., 2011)
+at 2x2x2mm resolution.
+
```

Comment:
I would add in something like: voxel-wise ALE scores were computed as the union of the MA maps. probably include something like that in ALE too. just something to add between gaussian kernels and thresholding the ALE map.